### PR TITLE
refactor: remaining code migrated to null safety

### PR DESCRIPTION
**Pull request by amanv8060**
refactor: remaining code migrated to null safety leaving tests to be migrated

part of #527 

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -73,7 +71,7 @@ class GalleryApp extends StatelessWidget {
             supportedLocales: GalleryLocalizations.supportedLocales,
             locale: GalleryOptions.of(context).locale,
             localeListResolutionCallback: (locales, supportedLocales) {
-              deviceLocale = locales.first;
+              deviceLocale = locales!.first;
```
Comment (guidezpl) [inline lib/main.dart:74]:
```suggestion
              deviceLocale = locales?.first;
```

**Comment by guidezpl** [comment]
LGTM! Is there anything else to do?

**Review by guidezpl — APPROVED** [review]

**Comment by amanv8060** [comment]
no, will migrate the tests this week as well.

**Comment by guidezpl** [comment]
thanks you so much!

**Comment by guidezpl** [comment]
all works fine with https://github.com/flutter/gallery/pull/625, I went through the whole app and only found one other bug: if you open settings in the upper right, selecting a different text scaling option. Do you mind taking a look?

**Pull request merged**